homeassistant/components/light/zwave.py

```diff
@@ -117,10 +131,46 @@ def turn_on(self, **kwargs):
         # brightness.
         brightness = (self._brightness / 255) * 99
 
+        # If 0, then just send command to turn on light immediately. Some dimmer
+        # switches will still have a fade-on period that's built into the
+        # hardware.
+        transition = kwargs.get(ATTR_TRANSITION, 0)
```

Comment:
If transition is not given in turn_on, turn it on and escape from this function. Will make it less error prone.